deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -0,0 +1,213 @@
+import deeplake
+from deeplake.core.vectorstore import utils, vector_search
+
+import logging
+import uuid
+from functools import partial
+from typing import Optional, Any, Iterable, List, Dict, Callable
+
+import numpy as np
+
+
+logger = logging.getLogger(__name__)
+
+
+class DeepLakeVectorStore:
+    """Base class for DeepLakeVectorStore"""
+
+    _DEFAULT_DEEPLAKE_PATH = "./deeplake_vector_store"
+
+    def __init__(
+        self,
+        dataset_path: str = _DEFAULT_DEEPLAKE_PATH,
+        token: Optional[str] = None,
+        embedding_function: Optional[callable] = None,
+        read_only: Optional[bool] = False,
+        ingestion_batch_size: int = 1024,
+        num_workers: int = 4,
+        exec_option: str = "python",
+        **kwargs: Any,
+    ) -> None:
+        self.ingestion_batch_size = ingestion_batch_size
+        self.num_workers = num_workers
+        creds = {"creds": kwargs["creds"]} if "creds" in kwargs else {}
+        self.dataset = utils.create_or_load_dataset(
+            dataset_path, token, creds, logger, read_only, **kwargs
+        )
+        self._embedding_function = embedding_function
+        self.exec_option = exec_option
+
+    def add(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        embeddings: Optional[Any] = None,
+    ) -> List[str]:
+        """Adding elements to deeplake vector store
+
+        Args:
+            texts (Iterable[str]): texts to add to deeplake vector store
+            metadatas (Optional[List[dict]], optional): List of metadatas.. Defaults to None.
+            ids (Optional[List[str]], optional): List of document IDs. Defaults to None.
+            embeddings (Optional[Any], optional): embedding of texts. Defaults to None.
+        Returns:
+            ids (List[str]): List of document IDs
+        """
+        elements = self._create_elements(ids, texts, metadatas, embeddings)
+        self._run_data_injestion(elements)
```

Comment:
IMO run injection would also have remote or local later, though no need to worry about this now.